File: backend/app/schemas/video.py
import uuid
from typing import Optional, Literal
from pydantic import BaseModel, ConfigDict


class VideoClipRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    project_id: uuid.UUID
    segment_id: Optional[uuid.UUID] = None
    sequence_order: int
    source_type: Literal["generated", "asset", "uploaded"]
    status: Literal["pending", "generating", "ready", "failed"]
    prompt: Optional[str] = None
    video_url: Optional[str] = None
    duration_ms: Optional[int] = None
    trim_start_ms: Optional[int] = None
    trim_end_ms: Optional[int] = None
    timeline_start_ms: Optional[int] = None
    timeline_end_ms: Optional[int] = None


class VideoClipUpdate(BaseModel):
    """Partial update for editor edits (prompt, trim, ordering)."""
    prompt: Optional[str] = None
    trim_start_ms: Optional[int] = None
    trim_end_ms: Optional[int] = None
    sequence_order: Optional[int] = None


class VideoBrief(BaseModel):
    """Campaign-level visual direction stored in Project.video_brief."""
    style_prompt: Optional[str] = None
    character_sheet: Optional[str] = None
    video_master_url: Optional[str] = None


class SubtitleStyle(BaseModel):
    """Stored in Project.subtitle_style; controls ASS burn-in."""
    enabled: bool = True
    font_size: int = 36
    position: Literal["bottom", "top", "center"] = "bottom"
    color: str = "FFFFFF"


class VideoClipsResponse(BaseModel):
    """Everything the studio's Video tab needs in one read."""
    clips: list[VideoClipRead] = []
    video_brief: Optional[dict] = None
    subtitle_style: Optional[dict] = None

File: frontend/types/video.ts
export type CampaignMedium = 'audio' | 'video';

export type StudioTab = 'audio' | 'video';

export type VideoClipStatus = 'pending' | 'generating' | 'ready' | 'failed';

export type VideoSourceType = 'generated' | 'asset' | 'uploaded';

export interface VideoClip {
    id: string;
    project_id: string;
    segment_id?: string | null;
    sequence_order: number;
    source_type: VideoSourceType;
    status: VideoClipStatus;
    prompt?: string | null;
    video_url?: string | null;
    duration_ms?: number | null;
    trim_start_ms?: number | null;
    trim_end_ms?: number | null;
    timeline_start_ms?: number | null;
    timeline_end_ms?: number | null;
}

export interface VideoBrief {
    style_prompt?: string | null;
    character_sheet?: string | null;
    video_master_url?: string | null;
}

export interface SubtitleStyle {
    enabled: boolean;
    font_size: number;
    position: 'bottom' | 'top' | 'center';
    color: string;
}

export const DEFAULT_SUBTITLE_STYLE: SubtitleStyle = {
    enabled: true,
    font_size: 54,
    position: 'bottom',
    color: 'FFFFFF',
};

export type VideoExportStatus = 'idle' | 'running' | 'ready' | 'failed';

export interface VideoExportState {
    status: VideoExportStatus;
    url: string | null;
    error: string | null;
}

export const IDLE_EXPORT: VideoExportState = { status: 'idle', url: null, error: null };
